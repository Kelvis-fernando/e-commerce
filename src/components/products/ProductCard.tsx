import { useModal } from "../../hooks/layout/useModal";
import {
  CardAction,
  CardButton,
  ProductCardContainer,
} from "../../styles/products/productCard";
import { ProductCardProps } from "../../types/productCardProps";
import Modal from "../layout/Modal";
import Quantity from "../layout/Quantity";
import ProductExpandedModal from "./ProductExpandedModal";

const ProductCard = ({
  id,
  name,
  brand,
  image,
  price,
  description,
  product,
}: ProductCardProps) => {
  const { handleModalClose, isModalOpen, setIsModalOpen } = useModal();

  return (
    <>
      <ProductCardContainer>
        <div onClick={() => setIsModalOpen(true)}>
          <img src={image} alt="product" />
          <h1>{name}</h1>
          <h2>
            <span>$ {price}</span>
            <span>{brand}</span>
          </h2>
        </div>
        <CardAction>
          <CardButton>Comprar</CardButton>
          <Quantity />
        </CardAction>
      </ProductCardContainer>
      <Modal isOpen={isModalOpen} onClose={handleModalClose}>
        <ProductExpandedModal product={product} />
      </Modal>
    </>
  );
};

export default ProductCard;
