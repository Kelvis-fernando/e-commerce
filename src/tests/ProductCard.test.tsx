import React from "react";
import { render } from "@testing-library/react";
import ProductCard from "../components/products/ProductCard";

describe("ProductCard", () => {
  test("should render and match snapshot", () => {
    const { container } = render(<ProductCard />);

    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
