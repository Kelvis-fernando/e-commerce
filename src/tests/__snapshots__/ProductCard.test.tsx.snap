// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard should render and match snapshot 1`] = `
<div>
  <div
    class="sc-gswNZR kNlMOH"
  >
    <div>
      <img
        alt="product"
        src=""
      />
      <h1 />
      <h2>
        <span>
          R$ 
        </span>
        <span />
      </h2>
    </div>
    <div
      class="sc-dkrFOg dKirdS"
    >
      <div
        class="sc-bcXHqe hHqvzg"
      >
        <button>
          +
        </button>
        <div>
          1
        </div>
        <button>
          -
        </button>
      </div>
      <button
        class="sc-hLBbgP lkCtGZ"
      >
        Comprar
      </button>
    </div>
  </div>
</div>
`;
